src/main/java/seedu/duke/manager/Manager.java

```diff
@@ -0,0 +1,103 @@
+package seedu.duke.manager;
+
+import seedu.duke.command.BadCommandException;
+import seedu.duke.command.Command;
+import seedu.duke.command.CommandParser;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+
+/**
+ * Manager is the superclass for all WellNUS++ event drivers <br>
+ * <br>
+ * Each manager is in charge of 'managing' exactly one feature. <br>
+ * For example, <i>hb</i> and <i>reflect</i>. <br>
+ * <br>
+ * The manager should run an event driver (infinite loop) and is in charge
+ * of a Feature's input, output, 'business' logic and graceful termination
+ */
+public abstract class Manager {
+
+    private final CommandParser commandParser;
+    private ArrayList<Command> argumentList;
+
+    /**
+     * Construct a feature Manager to handle control flow for the given feature.
+     */
+    public Manager() {
+        this.commandParser = new CommandParser();
+    }
+
+    /**
+     * Utility function to get the CommandParser tied to the Manager class.
+     *
+     * @return commandParser reference to this manager's instance of CommandParser
+     */
+    public CommandParser getCommandParser() {
+        return this.commandParser;
+    }
+
+    /**
+     * Utility function to get the featureName this Manager is administering
+     *
+     * @return name of the feature that this Manager handles
+     */
+    public abstract String getFeatureName();
+
+    /**
+     * Utility function to get a summary description of the feature this Manager is administering
+     *
+     * @return summary description of the feature that this Manager handles
+     */
+    public abstract String getBriefDescription();
+
+    /**
+     * Utility function to get the full description of the feature this Manager is administering
+     *
+     * @return full description of the feature that this Manager handles
+     */
+    public abstract String getFullDescription();
+
+    /**
+     * Utility function to set a list of main arguments the feature supports
+     */
+    public abstract void setArgumentList();
+
+    /**
+     * Utility function to get a list of main arguments the feature supports
+     *
+     * @return <code>ArrayList</code> of <code>Command</code>s
+     */
+    public ArrayList<Command> getArgumentList() {
+        return argumentList;
+    }
```

Comment:
Will this argumentList be used for 'help' function? In that case should we add a help method to be inherited by all manager-subclass?

Reply:
Good point. The help behaviour is being implemented by @haoyangw so I'll keep him in the loop for this.

Do we want to enforce a Manager-level help function that prints every command in the argumentList?
Or are we offloading this to a `HelpCommand`?

Implementation in manager would look something like (pseudocode): 

```java
public void printHelpMessage() {
    System.out.println("Usage for " + getFeatureName());
    for(Command cmd : commands) {
        Ui.print(cmd.toString());
    }
}
```

Reply:
Thank you for the notification.

Yes, I propose breaking `help` into 4 parts: the feature keyword(e.g. `hb`), a `String` of the entire list of supported commands, a brief description of the feature, and a detailed description of the feature.

I propose delegating the 4 parts this way: feature keyword is stored in the feature's `Manager`(exposed with a getter), `String` of all subcommands is also stored in the feature's `Manager`(also exposed with a getter), brief description of the feature is also stored in the corresponding `Manager`. The complicated part, and the part that warrants the most discussion is I believe the detailed description. This is my rough concept for it:

-`Manager` will have a detailed _overall_ description of the feature, such as `Atomic habits: Inspired by the book ..., this feature allows you to propose and keep track of atomic habits you desire to develop in the hopes that it may improve your lifestyle.` for the atomic habits feature
-Every `Command` of a feature will provide(via getters, one each):
1. `String` of all the arguments for that specific `Command`
2. Detailed description of the command(what it does, any default behaviour when optional arguments are omitted, etc)
-`Manager` will provide a getter for a detailed description of the entire feature's commands. This is essentially just iterating through all of the `Command`s and calling their argument getter and detailed description getter
-`HelpCommand`'s job is to standardise the format of the brief description and detailed descriptions, and iterate through all the `Manager`s when printing the detailed description(via each `Manager`'s getter), and print the output using `TextUi`
-`HelpCommand` **will not** know about what commands/features are supported by `WellNus` or what description they have, this will be delegated to helper methods provided by `Manager`s for encapsulation and to aid maintainability so that individual feature developers will decide how they want to describe their features and commands

What are your thoughts @YongbinWang @nichyjt ?

Reply:
In the context of my above suggestions, I propose making getArgumentList() `abstract`, and returning a `String` instead. That might entail renaming the method to something like `getStringOfArguments`

Reply:
Agree for the most part!

Rather than parsing it as one gigantic string, I think returning it as an arraylist of strings might be better, especially for the formatting HelpUi will be doing